benchmark/solver/solver.cpp

```diff
@@ -472,9 +475,9 @@ void solve_system(const std::string &solver_name,
 
         // timed run
         auto generate_timer = get_timer(exec, FLAGS_gpu_timer);
-        auto apply_timer = get_timer(exec, FLAGS_gpu_timer);
-        for (unsigned int i = 0; i < FLAGS_repetitions; i++) {
-            auto x_clone = clone(x);
+        auto x_clone = clone(x);
+        for (auto status : ic.run(false)) {
+            x_clone = clone(x);
 
             exec->synchronize();
             generate_timer->tic();
```

Comment:
First thing: you can already use `ic.compute_average_timings` to get the output, since `ic` has internally a copy of the shared ptr for the timer (but only for the apply timer). However, that seems a bit awkward. 

From my viewpoint, IC should not handle any timings in this instance, as `manage_timings` is set to `false`. Adding timings functions to IC would weaken the distinction to the managed case. If the non-managed IC run is requested, the user should take care of the timings.

Reply:
It makes sense from this point.
My point is iteration control needs the results from the corresponding timer, so it can be wrong usage.
but I also realize that may make the tic/toc from TimerManager different?
maybe we need to add some comment about that need to use the corresponding timer such that the iteration control can really check it?

Reply:
Perhaps I could add a `get_timer` to IC. That should help to use the correct timer. Otherwise, I can't think of a graceful way of adding tic/toc directly to IC.

Reply:
Yeah, it is good.